### PR TITLE
Switch to tc pulse

### PR DESCRIPTION
**Pull request by owlishDeveloper**
So I am using yarn 1.9.4, and all the integrity checks are gone from the `yarn.lock` :( Do I upgrade my yarn or what do I do? 

Also, I removed all the beautiful indentation on purpose - I see that it's not being maintained as the codebase matures, and already looks mishmash. So I decided to go ahead and remove it (I think maintaining it can be hard and doesn't seem like something we can automate - correct me if I'm wrong -, and it doesn't improve readibility _that_ much). Let me know if that's really bad :)

### CODE REVIEW

**Comment by djmitche** [comment]
Regarding yarn, I saw a similar thing.  Yarn usually prints a line you can copy-paste to upgrade it.  Or just use `npm upgrade -U -g yarn`.

**Comment by djmitche** [comment]
Removing the aligning spaces is great, but it makes review easier if that's done in a distinct, whitespace-only commit.  Then I can skim that commit and see that it's whitespace-only, and review the actual important changes in a separate commit.

File: src/handlers.js [hunk 1]
```diff
@@ -22,76 +23,55 @@ var helpers     = require('./helpers');
...
-  this.listener         = null;
+  this.IndexedTask = options.IndexedTask;
+  this.Namespace = options.Namespace;
+  this.queue = options.queue;
+  this.queueEvents = options.queueEvents;
+  this.credentials = options.credentials;
+  this.routePrefix = options.routePrefix;
+  this.queueName = options.queueName;  // Optional
+  this.monitor = options.monitor;
+  this.pc = null;
```
Comment (djmitche) [inline src/handlers.js]: pq?

File: src/handlers.js [hunk 2]
```diff
@@ -22,76 +23,55 @@ var helpers     = require('./helpers');
...
-  // Binding for completed tasks
-  var completedBinding = this.queueEvents.taskCompleted(
-    'route.' + this.routePrefix + '.#'
+  this.pc = await consume(
+    {
+      client: this.pulseClient,
+      bindings: [this.binding],
+      queueName: this.queueName,
+    },
+    message => this.monitor.timedHandler('listener', this.messageHandler(message)),
```
Comment (djmitche) [inline src/handlers.js]: I think this has the same issue as described in taskcluster-github.  The call to `this.messageHandler(message)` will be executed and its return value given to `this.monitor.timedHandler`, which will then treat it as taking no time at all to execute.

The way I find most clear to write this is:

```javascript
message => this.monitor.timedHandler('listener', () => this.messageHandler(message)),
```
which is a function taking one argument (message), which calls timedHandler with a handler name ('listener') and a function which takes no arguments and calls `this.messageHandler(message)`.  timedHandler will then call this last function and time how long it takes to execute.

File: src/main.js [hunk 3]
```diff
@@ -132,9 +132,27 @@ var load = loader({
...
 
+  pulseClient: {
+    requires: ['cfg', 'monitor'],
+    setup: ({cfg, monitor}) => {
+      const {namespace, expiresAfter, contact} = cfg.pulse;
+      const {rootUrl, credentials} = cfg.taskcluster;
+      return new Client({
+        namespace,
+        monitor,
+        credentials: claimedCredentials({
```
Comment (djmitche) [inline src/main.js]: Like in tc-github, let's change this back to use `pulseCredentials` with username/password/hostname/vhost.

File: src/handlers.js (hunk 1 above)
Comment (owlishDeveloper) [reply src/handlers.js]: yep!

File: src/handlers.js (hunk 2 above)
Comment (owlishDeveloper) [reply src/handlers.js]: Investigating this

File: src/handlers.js [hunk 4]
```diff
@@ -22,76 +23,55 @@ var helpers     = require('./helpers');
...
+  this.Namespace = options.Namespace;
+  this.queue = options.queue;
+  this.queueEvents = options.queueEvents;
+  this.credentials = options.credentials;
+  this.routePrefix = options.routePrefix;
+  this.queueName = options.queueName;
+  this.monitor = options.monitor;
+  this.pq = null;
+  // Binding for completed tasks
+  this.binding = options.queueEvents.taskCompleted('route.' + options.routePrefix + '.#');
```
Comment (imbstack) [inline src/handlers.js]: Can we do this with a template string? I just find them a bit more readable.

**Comment by djmitche** [comment]
deployed!  I had to add `pulse: namespace: "taskcluster-index"` to config.yml to get it to start up.

**Pull request merged**